lib/iris/experimental/regrid.py

```diff
@@ -994,14 +1072,78 @@ def _regrid_indices(cells, depth, points):
     # contribution from one or more source points.
     rows = np.nonzero(sum_weights)
 
-    # Calculate the numerator of the weighted mean (M, 1).
-    numerator = sparse_matrix * src_cube.data.reshape(-1, 1)
+    # NOTE: when source points are masked, this 'sum_weights' is possibly
+    # incorrect and needs re-calculating.  Likewise 'rows' may cover target
+    # cells which happen to get no data.  This is dealt with by adjusting as
+    # required in the '__perform' function, below.
+
+    regrid_info = (sparse_matrix, sum_weights, rows, grid_cube)
+    return regrid_info
```

Comment:
I'd rather not as I'm seeing "regrid_info" as a black box : Odd context returned by one routine, to be passed unchanged to the other.
Just to prove it, I have actually changed what was in there several times during development !